common/achievement/evaluate.ts

```diff
@@ -48,22 +49,18 @@ export async function evaluateAchievement(
         const aggFunction = aggregators[aggregator].function;
 
         if (!bucketCreatorFunction || !bucketAggregatorFunction || !aggFunction) {
-            return;
+            logger.error(
+                `No bucket creator or aggregator function found for ${bucketCreator}, ${aggregator} or ${bucketAggregator} during the evaluation of achievement`
+            );
+            continue;
```

Comment:
I would do a return in this case. It doesn't make sense to evaluate if we cannot get all the data needed.